spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageLocation.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.springframework.util.Assert;
+
+/**
+ * Represents a Google Cloud Storage location provided by the user.
+ *
+ * @author Daniel Zou
+ */
+public final class GoogleStorageLocation {
+
+	private static final String GCS_URI_FORMAT = "gs://%s/%s";
+
+	private final String bucketName;
+
+	private final String blobName;
+
+	private final URI uri;
+
+	/**
+	 * Constructs a {@link GoogleStorageLocation} based on the provided Google Storage URI string.
+	 * The URI string is of the form: {@code gs://<BUCKET_NAME>/<PATH_TO_FILE>}
+	 *
+	 * @param gcsLocationUriString a Google Storage URI string to a bucket/folder/file.
+	 */
+	public GoogleStorageLocation(String gcsLocationUriString) {
+		try {
+			URI locationUri = new URI(gcsLocationUriString);
+
+			Assert.isTrue(
+					gcsLocationUriString.startsWith("gs://"),
+					"A Google Storage URI must start with gs://");
+
+			Assert.isTrue(
+					locationUri.getAuthority() != null,
+					"No bucket specified in the location: " + gcsLocationUriString);
+
+			this.bucketName = locationUri.getAuthority();
+			this.blobName = getBlobPathFromUri(locationUri);
+
+			// ensure that if it's a bucket handle, location ends with a '/'
+			if (this.blobName == null && !gcsLocationUriString.endsWith("/")) {
+				locationUri = new URI(gcsLocationUriString + "/");
+			}
+			this.uri = locationUri;
+		}
+		catch (URISyntaxException e) {
+			throw new IllegalArgumentException(
+					"Invalid location: " + gcsLocationUriString);
+		}
+	}
+
+	/**
+	 * Returns the Google Storage bucket name.
+	 */
+	public String getBucketName() {
+		return bucketName;
+	}
+
+	/**
+	 * Returns the path to the blob/folder relative from the bucket root. Returns the empty
```

Comment:
It would return `null` and not an empty string, based on `getBlobPathFromUri()` behavior.

Reply:
Done.